### PR TITLE
Fix php8 warnings

### LINKED ISSUE #83: PHP8.1 Warning
After switch to PHP8.1

PHP Warning:  Undefined array key "showCount" in <xxx>/wiki/lib/plugins/cloud/syntax.php on line 138

### PR DESCRIPTION
**Pull request by dregad**
Fixes #83

### CODE REVIEW

**Comment by Klap-in** [comment]
Thanks for the PR.
Could you please set on line 41 the flag array with defaults? I prefer to have defined variables much as possible.
So in this case that is just one flag if I see it right:
```
$flags=[
    'showCount' => false
];
```

**Comment by Klap-in** [comment]
Ah, I see double use of the flags variable, should be line 45.

**Comment by dregad** [comment]
Will fix

**Comment by Klap-in** [comment]
Might be useful to fix some php notices as well: all explodes with a limit e.g.`explode('>', …, 2) ` could be wrapped with array_pad(…,2,'')

**Comment by dregad** [comment]
I fixed some with a different approach, but using array_pad may be cleaner, let me know what you prefer

File: syntax.php [hunk 1]
```diff
@@ -36,9 +36,16 @@ function handle($match, $state, $pos, Doku_Handler $handler) {
...
-        list($junk, $num) = explode(':', $num, 2);
-        $flags = null;
+        // Add trailing separator to ensure we always have 2 entries in the array
+        list($num, $ns) = explode('>', $match . '>', 2);
+        $ns = rtrim($ns, '>');
+        list($junk, $num) = explode(':', $num . ':', 2);
+        $num = rtrim($num, ':');
+
+        // Set default flag values
+        $flags = [
```
Comment (Klap-in) [inline syntax.php:46]: $flags is also fed to the preg_match, so you should set this later at line 52, not here. The check on line 145 is then not needed anymore.

I prefer the other way, but of course this works for the explode-list combi as well.

File: syntax.php [hunk 2]
```diff
@@ -56,6 +52,12 @@ function handle($match, $state, $pos, Doku_Handler $handler) {
                 }
             }
         }
+        // Set default flag values
+        if (!$flags) {
+            $flags = [
+                'showCount' => false,
+            ];
+        }
```
Comment (Klap-in) [inline syntax.php:60]: Sorry, using line numbers was confusing, this what I tried to say. Thanks for the effort.
```Diff
if (preg_match ('/\[.*\]/', $junk, $flags) === 1) {
             $flags = trim ($flags [0], '[]');
             $found = explode(',', $flags);
-            $flags = array();
+             $flags = [
+                 'showCount' => false,
+             ];
            foreach ($found as $flag) {
                if (in_array($flag, $this->knownFlags)) {
                    // Actually we just set flags as present
                    // Later we might add values to flags like key=value pairs
                    $flags [$flag] = true;
                 }
             }
         }
-         // Set default flag values
-       if (!$flags) {
-             $flags = [
-                 'showCount' => false,
-             ];
-         }
         


```

Comment (dregad) [reply syntax.php:60]: Now that I removed the extra check at line 145 per our earlier discussion, initializing $flags before the foreach as you suggest would expose us to another undefined array key warning, if preg_match() didn't match.

If your intention is to ensure the code is forward-compatible with future new flags, then I would propose to initialize the $flags array before the `if (preg_match` (like I did in the previous commit), and use another variable for preg_match() matches.

```php
$flags = [
    'showCount' => false,
];
if (preg_match('/\[.*\]/', $junk, $matches) === 1) {
    $matches = trim($matches[0], '[]');
    $found = explode(',', $matches);
    $flags = array();
    foreach ($found as $flag) {
        if (in_array($flag, $this->knownFlags)) {
            // Actually we just set flags as present
            // Later we might add values to flags like key=value pairs
            $flags [$flag] = true;
        }
    }
}
```

Let me know what you think, before I commit.

Comment (Klap-in) [reply syntax.php:60]: Ah, sure! I agree

Comment (dregad) [reply syntax.php]: OK, change has been committed.

**Comment by Klap-in** [comment]
Thank you!

**Pull request merged**